File: server/SmartGeoIot/Models/Project.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;

namespace SmartGeoIot.Models
{
    public enum ProjectCode
    {
        [EnumMember(Value = "B982-C")]
        B972,
        
        [EnumMember(Value = "B982-S")]
        B982_S,
        
        [EnumMember(Value = "Hidroleg")]
        Hidroleg,
        
        [EnumMember(Value = "DJRFleg")]
        DJRFleg,
        
        [EnumMember(Value = "AM-1leg")]
        AM_1leg,
        
        [EnumMember(Value = "FluxoRDleg")]
        FluxoRDleg,
        
        [EnumMember(Value = "B972-P")]
        B972_P,
        
        [EnumMember(Value = "B981")]
        B981,

        [EnumMember(Value = "B978")]
        B978,

        [EnumMember(Value = "B987")]
        B987
    }

    public class Project
    {
        [Key, Column(TypeName = "char(36)"), MaxLength(36)]
        public string ProjectUId { get; set; }

        [MaxLength(200)]
        public string Name { get; set; }

        [MaxLength(400)]
        public string Description { get; set; }

        [MaxLength(50)]
        public string Code { get; set; }
    }

    public class ProjectDevice
    {
        [Key]
        public Guid Id { get; set; }

        [MaxLength(50)]
        public string Code { get; set; }
        public string DeviceId { get; set; }
    }
}

File: client.Angular/src/app/smart-geo-iot/project.ts
export class Project {
    projectUId: string;
    name: string;
    description: string;
    code: string;

    get getUrlReportLink(): string {
        if (this.name.toLowerCase() === 'djrf') {
            return 'djrf';
        }
        if (this.name.toLowerCase() === 'aguamon') {
            return 'aguamon';
        }
        if (this.name.toLowerCase() === 'aguamon-2') {
            return 'aguamon2';
        }
        if (this.code === ProjectEnum.B972_P) {
            return 'trm';
        }
        // if (this.name.toLowerCase() === 'tsp') {
        //     return 'tsp';
        // }
        if (this.code === ProjectEnum.B981) {
            return ('tqa');
        }
        if (this.code === ProjectEnum.B972) {
            return ('clampon');
        }
        if (this.code === ProjectEnum.B982_S) {
            return ('tsp');
        }
        if (this.code === ProjectEnum.B978) {
            return ('b978');
        }
        if (this.code === ProjectEnum.B987) {
            return ('b987');
        }
        if (this.code === ProjectEnum.B980) {
            return ('b980');
        }
        if (this.code === ProjectEnum.B975) {
            return ('b975');
        }
        return 'hidro';
    }

    get getURLDashboardLink(): string {
        if (this.code === ProjectEnum.B975) {
            return ('radiodados/dashboard-b975');
        }

        return 'radiodados/dashboard';
    }
}

export class ProjectEnum {
    static readonly B972 = 'B982-C';
    static readonly B982_S = 'B982-S';
    static readonly Hidroleg = 'Hidroleg';
    static readonly DJRFleg = 'DJRFleg';
    static readonly AM_1leg = 'AM-1leg';
    static readonly FluxoRDleg = 'FluxoRDleg';
    static readonly B972_P = 'B972-P';
    static readonly B981 = 'B981';
    static readonly B978 = 'B978';
    static readonly B987 = 'B987';
    static readonly B980 = 'B980';
    static readonly B975 = 'B975';
}
